src/api/client.ts

```diff
@@ -0,0 +1,744 @@
+	type CreateFeatureFlagInput,
+	CreateFeatureFlagInputSchema,
+	type UpdateFeatureFlagInput,
+	UpdateFeatureFlagInputSchema,
+	FeatureFlagSchema,
+} from "../schema/flags";
+import { ApiPropertyDefinitionSchema } from "../schema/api";
+import { PropertyDefinitionSchema } from "../schema/properties";
+import {
+	type CreateInsightInput,
+	CreateInsightInputSchema,
+	type ListInsightsData,
+	ListInsightsSchema,
+} from "../schema/insights";
+import {
+	type CreateDashboardInput,
+	CreateDashboardInputSchema,
+	type ListDashboardsData,
+	ListDashboardsSchema,
+} from "../schema/dashboards";
+import { z } from "zod";
+
+export type Result<T, E = Error> = { success: true; data: T } | { success: false; error: E };
+
+export interface ApiConfig {
+	apiToken: string;
+	baseUrl?: string;
+}
+export class ApiClient {
+	private config: ApiConfig;
+	private baseUrl: string;
+
+	constructor(config: ApiConfig) {
+		this.config = config;
+		this.baseUrl = config.baseUrl || BASE_URL;
+	}
+	private buildHeaders() {
+		return {
+			Authorization: `Bearer ${this.config.apiToken}`,
+			"Content-Type": "application/json",
+		};
+	}
+
+	private async fetchWithSchema<T>(
+		url: string,
+		schema: z.ZodType<T>,
+		options?: RequestInit,
+	): Promise<Result<T>> {
+		try {
+			const response = await fetch(url, {
+				...options,
+				headers: {
+					...this.buildHeaders(),
+					...options?.headers,
+				},
+			});
+
+			if (!response.ok) {
+				if (response.status === 401) {
+					throw new Error(ErrorCode.INVALID_API_KEY);
+				}
+
+				try {
+					const errorData = (await response.json()) as any;
+					if (errorData.type === "validation_error" && errorData.code) {
+						throw new Error(`Validation error: ${errorData.code}`);
+					}
+				} catch {}
+
+				throw new Error(`Request failed: ${response.statusText}`);
+			}
+
+			const rawData = await response.json();
+			const parseResult = schema.safeParse(rawData);
+
+			if (!parseResult.success) {
+				throw new Error(`Response validation failed: ${parseResult.error.message}`);
+			}
+
+			return { success: true, data: parseResult.data };
+		} catch (error) {
+			return { success: false, error: error as Error };
+		}
+	}
+
+	organizations() {
+		return {
+			list: async (): Promise<Result<Organization[]>> => {
+				const responseSchema = z.object({
+					results: z.array(OrganizationSchema),
+				});
+
+				const result = await this.fetchWithSchema(
+					`${this.baseUrl}/api/organizations/`,
+					responseSchema,
+				);
+
+				if (result.success) {
+					return { success: true, data: result.data.results };
+				}
+				return result;
+			},
+
+			get: async ({ orgId }: { orgId: string }): Promise<Result<Organization>> => {
+				return this.fetchWithSchema(
+					`${this.baseUrl}/api/organizations/${orgId}/`,
+					OrganizationSchema,
+				);
+			},
+
+			projects: ({ orgId }: { orgId: string }) => {
+				return {
+					list: async (): Promise<Result<Project[]>> => {
+						const responseSchema = z.object({
```

Comment:
You're building this `results` object in several places, a helper would make sense